nidirect_gp/src/Controller/GpSearchController.php

```diff
@@ -233,6 +261,23 @@ public function handleSearchRequest() {
 
   }
 
+  /**
+   * Update the Google maps api key if necessary.
+   */
+  private function updateApiKey() {
+    // Ensure that the geocoder provider api key is correct (as the api key
+    // cannot be held in config, it may be necessary to update it here with
+    // the api key that is held in the environment variable).
+    $googlemap_config = $this->configFactory->getEditable('geocoder.geocoder_provider.googlemaps')->get('configuration');
+    if (!empty($googlemap_config) &&
+      isset($googlemap_config['apiKey']) &&
+      ($googlemap_config['apiKey'] != getenv('GOOGLE_MAP_API_SERVER_KEY'))) {
+      // Overwrite the google map api key.
+      $googlemap_config['apiKey'] = getenv('GOOGLE_MAP_API_SERVER_KEY');
+      $this->configFactory->getEditable('geocoder.geocoder_provider.googlemaps')->set('configuration', $googlemap_config)->save();
```

Comment:
This is not ideal but is necessary to avoid holding an api key in config.
It has been optimised as much as possible, so only the first GP search will actually do this update after each config import.